qrand/platforms/qsharp/job.py

```diff
@@ -88,18 +85,14 @@ def num_measurements(self, num_measurements: Optional[int]) -> None:
                 Using max_measurements instead.",
                 UserWarning,
             )
-            num_measurements = self.backend.max_measurements
-        self._shots, self._experiments = compute_bounded_factorization(
-            num_measurements,
-            self.backend.max_shots,
-            self.backend.max_experiments,
-        )
+            self._num_measurements = self.backend.max_measurements
```

Comment:
Watch out, this does not make sense, you need to understand what you are doing.

In this property setter we are assigning the value of `self._num_measurements`. The if statement makes sure that if we try to set it to something higher than the max allowed, we actually just set it to that maximum (and warns the user accordingly). But if that is not the case we need to assign a value as well!

What about this?

```python
if self.backend.max_measurements < num_measurements:            
    warn(                
        f"Number of measurements unsupported by the job's Backend: \                
        {self.backend.max_measurements}&lt;{num_measurements}. \                
        Using max_measurements instead.",                
        UserWarning,            
    )            
    num_measurements = self.backend.max_measurements        
self._num_measurements = self.backend.max_measurements
```

Reply:
Watch out, I had a typo in my code. I just fixed it!